### PR TITLE
Add tech.keyforge.keyforge_master

### PR DESCRIPTION
**Pull request by 5t4r-10rd**
### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
KeyForge Master is a configurator for KeyForge devices (macropads). You can add predefined shortcuts to the buttons and encoders. Also you you can create custom actions/macros.

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: icons/256x256.png
Comment (hfiguiere) [inline icons/256x256.png:1]: remove this you don't need.

File: icons/512x512.png
Comment (hfiguiere) [inline icons/512x512.png:1]: remove this you don't need.

File: icons/64x64.png
Comment (hfiguiere) [inline icons/64x64.png:1]: remove this you don't need it.

File: tech.keyforge.master.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,85 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>tech.keyforge.master</id>
+  <developer id="tld.vendor">
```
Comment (hfiguiere) [inline tech.keyforge.master.metainfo.xml:4]: this is an INVALID developer ID

File: tech.keyforge.master.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,85 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>tech.keyforge.master</id>
```
Comment (hfiguiere) [inline tech.keyforge.master.metainfo.xml:3]: "Master" doesn't seem to be the app name according to the `<name>` tag.

File: tech.keyforge.master.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    <keyword>macro</keyword>
+
+  </keywords>
+
+  <branding>
+    <color type="primary" scheme_preference="light">#d2d6d9</color>
+    <color type="primary" scheme_preference="dark">#5e6567</color>
+  </branding>
+
+  <icon type="stock">tech.keyforge.master</icon>
```
Comment (hfiguiere) [inline tech.keyforge.master.metainfo.xml:36]:
```suggestion
```

File: tech.keyforge.master.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      - install -Dm 644 tech.keyforge.master.metainfo.xml /app/share/metainfo/tech.keyforge.master.metainfo.xml
+      - install -Dm 755 run.sh /app/bin/run.sh
+      - install -Dm 644 64x64.png /app/share/icons/hicolor/64x64/apps/tech.keyforge.master.png
+      - install -Dm 644 256x256.png /app/share/icons/hicolor/256x256/apps/tech.keyforge.master.png
+      - install -Dm 644 512x512.png /app/share/icons/hicolor/512x512/apps/tech.keyforge.master.png
+      - install -Dm 644 scalable.svg /app/share/icons/hicolor/scalable/apps/tech.keyforge.master.svg
+
+    sources:
+      - type: file
+        url: https://master.keyforge.tech/flatpak-releases/keyforge-master_0.5_linux.tar.xz
```
Comment (hfiguiere) [inline tech.keyforge.master.yml:33]: how does that work on aarch64?

File: tech.keyforge.master.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  <developer id="tld.vendor">
+    <name>KeyForge</name>
+  </developer>
+  <name>KeyForge Master</name>
+  <summary>Configurator software for KeyForge devices</summary>
+  <url type="homepage">https://keyforge.tech/</url>
+  <url type="help">https://keyforge.tech/setup</url>
+
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>LicenseRef-proprietary=https://keyforge.tech/end-user-agreement/</project_license>
```
Comment (hfiguiere) [inline tech.keyforge.master.metainfo.xml:13]: is the binary redistributable? If there is no explicit permission, `extra-data` shall be used.

File: tech.keyforge.master.yml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  - "--share=ipc"
+  - "--share=network"
+  - "--socket=x11"
+
+modules:
+  - name: "keyforge-master"
+    buildsystem: "simple"
+    build-commands:
+      - mkdir /app/main
+      - tar -xf keyforge-master_0.5_linux.tar.xz -C /app/main --strip-components=1
```
Comment (bbhtt) [inline tech.keyforge.master.yml:21]: Use `type: archive` with `strip-components: 1` in the source and remove this.

File: tech.keyforge.master.yml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+      - rm  keyforge-master_0.5_linux.tar.xz
```
Comment (bbhtt) [inline tech.keyforge.master.yml:22]:
```suggestion
```

File: tech.keyforge.master.yml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      - install -Dm 644 tech.keyforge.master.desktop /app/share/applications/tech.keyforge.master.desktop
+      - install -Dm 644 tech.keyforge.master.metainfo.xml /app/share/metainfo/tech.keyforge.master.metainfo.xml
+      - install -Dm 755 run.sh /app/bin/run.sh
+      - install -Dm 644 64x64.png /app/share/icons/hicolor/64x64/apps/tech.keyforge.master.png
+      - install -Dm 644 256x256.png /app/share/icons/hicolor/256x256/apps/tech.keyforge.master.png
+      - install -Dm 644 512x512.png /app/share/icons/hicolor/512x512/apps/tech.keyforge.master.png
+      - install -Dm 644 scalable.svg /app/share/icons/hicolor/scalable/apps/tech.keyforge.master.svg
```
Comment (bbhtt) [inline tech.keyforge.master.yml:29]: Just the scalable icon is fine, you don't need the PNGs.

File: icons/256x256.png
Comment (5t4r-10rd) [reply icons/256x256.png:1]: Removed

File: icons/512x512.png
Comment (5t4r-10rd) [reply icons/512x512.png:1]: Removed

File: icons/64x64.png
Comment (5t4r-10rd) [reply icons/64x64.png:1]: Removed

File: tech.keyforge.master.metainfo.xml (hunk 1 above)
Comment (5t4r-10rd) [reply tech.keyforge.master.metainfo.xml]: developer ID removed

File: tech.keyforge.master.metainfo.xml (hunk 3 above)
Comment (5t4r-10rd) [reply tech.keyforge.master.metainfo.xml]: removed

File: tech.keyforge.master.yml (hunk 4 above)
Comment (5t4r-10rd) [reply tech.keyforge.master.yml:31]: I have no way to test it on aarch64, I have added `only-arches: - x86_64` for now.

File: tech.keyforge.master.yml (hunk 7 above)
Comment (5t4r-10rd) [reply tech.keyforge.master.yml]: line removed

File: tech.keyforge.master.yml (hunk 8 above)
Comment (5t4r-10rd) [reply tech.keyforge.master.yml]: I have removed the PNGs, thanks.

File: tech.keyforge.master.metainfo.xml (hunk 2 above)
Comment (hfiguiere) [reply tech.keyforge.master.metainfo.xml:3]: I'll rephrase: the app-id doesn't feel right for the above reason. The last component should represent the application name, which "Master" is not.

File: tech.keyforge.master.metainfo.xml (hunk 1 above)
Comment (hfiguiere) [reply tech.keyforge.master.metainfo.xml]: Removing it was NOT the request.

File: tech.keyforge.master.metainfo.xml (hunk 2 above)
Comment (5t4r-10rd) [reply tech.keyforge.master.metainfo.xml:3]: Ok, I will update the app ID

File: tech.keyforge.master.yml (hunk 4 above)
Comment (5t4r-10rd) [reply tech.keyforge.master.yml:31]: Yeah, you are right. I have added a flathub json file with `"only-arches": ["x86_64"]`

File: tech.keyforge.master.metainfo.xml (hunk 2 above)
Comment (5t4r-10rd) [reply tech.keyforge.master.metainfo.xml]: I have updated the app ID.

File: tech.keyforge.master.metainfo.xml (hunk 1 above)
Comment (5t4r-10rd) [reply tech.keyforge.master.metainfo.xml]: Ok, I get it. Now it should be fine. Thanks

File: tech.keyforge.master.metainfo.xml (hunk 5 above)
Comment (5t4r-10rd) [reply tech.keyforge.master.metainfo.xml]: I need to do some research on it. I will make the necessary changes tomorrow.

Comment (5t4r-10rd) [reply tech.keyforge.master.metainfo.xml]: > is the binary redistributable? If there is no explicit permission, `extra-data` shall be used.

Yes, the binary is redistributable. I have updated the EULA page accordingly.

Comment (bbhtt) [reply tech.keyforge.master.metainfo.xml]: > Redistribution: You may redistribute the binary form of the Software, unmodified, to any third party; however, you are prohibited from charging any fees for such redistribution or using such redistribution for any commercial gain.

**Comment by bbhtt** [comment]
bot, build tech.keyforge.keyforge_master

**Comment by bbhtt** [comment]
Some icons are missing here I think

image


And I don't have the hardware to test. Is it working?

**Comment by 5t4r-10rd** [comment]
> Some icons are missing here I think
> 
> image
> 
> And I don't have the hardware to test. Is it working?

There are no icons missing, it's just the keycap and the two knobs seen from above.
For now no one has the hardware, except us. We still have some work to do on the firmware side, it's not on the market.  I just wanted to launch the app on the stores a little bit earlier to check if everything is fine on this side.

**Comment by 5t4r-10rd** [comment]
I have installed the test build, everything looks good. Thank you guys.

**Comment by hfiguiere** [comment]
then let's hold back the merge until then.

**Comment by 5t4r-10rd** [comment]
I would need it on the store for the marketing side. If you could give me a reason it would be appreciated.

**Comment by 5t4r-10rd** [comment]
It is possible to be published as a beta app?

**Pull request closed without merging**